clippy_lints/src/unused_peekable.rs

```diff
@@ -70,15 +71,22 @@ impl<'tcx> LateLintPass<'tcx> for UnusedPeekable {
                     return;
                 }
 
+                let mut found_peek_call = false;
+
                 for stmt in &block.stmts[idx..] {
```

Comment:
Could simplify this for loop into `let found_peek_call = block.stmts[idx..].iter().any(|stmt| vis.visit_stmt(stmt).is_break());`